contracts/multisig-prover/src/execute.rs

```diff
@@ -39,8 +39,9 @@ pub fn construct_proof(
             let mut builder = CommandBatchBuilder::new(config.destination_chain_id, config.encoder);
 
             if let Some(new_worker_set) = new_worker_set {
-                save_next_worker_set(deps.storage, &new_worker_set)?;
-                builder.add_new_worker_set(new_worker_set)?;
+                if save_next_worker_set(deps.storage, &new_worker_set).is_ok() {
```

Comment:
I don't like the fact that here we assume any error returned by `save_next_worker_set` suggests a worker set rotation is in progress. It is already not entirely true, and it can easily become less true in the future.

Reply:
I changed it here: https://github.com/axelarnetwork/axelar-amplifier/commit/125f3dfc026141981669f780150f42478858668d

Used match to be able to handle more error types in future if needed.